packages/use-contractkit/src/contract-kit-reducer.ts

```diff
@@ -0,0 +1,128 @@
+import { Connector, Dapp, Network } from './types';
+import { localStorageKeys } from './constants';
+import { CONNECTOR_TYPES, UnauthenticatedConnector } from './connectors';
+
+export function contractKitReducer(
+  state: ReducerState,
+  action: Actions
+): ReducerState {
+  switch (action.type) {
+    case 'decrementPendingActionCount':
+      return {
+        ...state,
+        pendingActionCount: state.pendingActionCount - 1,
+      };
+
+    case 'setAddress':
+      if (action.payload) {
+        localStorage.setItem(localStorageKeys.lastUsedAddress, action.payload);
```

Comment:
The user of storage in a reducer is maybe okay for now but could be problematic later.
If we want this lib to serve other platforms, like react-native, storage writes will sometimes be `async`. A separate storage layer so that we're not directly hitting `localsStorage` may be a good idea but doesn't have to be part of this change. Just something to think about.

Reply:
That's a good point. I feel like I'd like to ruminate on this idea and look at what other people do in similar situations. My main concern was the desynchronization of memory state and local storage state, so I would be interested in still having some way to keep those linked.